frontend/packages/ux-editor/src/utils/exportUtils.ts

```diff
@@ -10,125 +10,143 @@ import type {
-      defaultLanguage,
-      (textResource) => textResource.id === textResourceBindings[textResourceBindingKey],
-    );
-    const textObject: ExportTextResource = {
-      id: textResourceBindings[textResourceBindingKey],
-      type: textResourceBindingKey,
-      text: textValues,
-    };
-    result.push(textObject);
-  });
-  return result;
-};
+    return exportComponent;
+  };
 
-const getTextValues = (
-  textResources: ITextResources,
-  textResourceKey: string,
-  defaultLanguage: string,
-  findFunction?: (textResource: ITextResource) => boolean,
-): ExportTextResourceValue[] => {
-  if (defaultLanguage) {
-    return [
-      {
-        language: defaultLanguage,
-        value: textResources[defaultLanguage].find(findFunction)?.value ?? textResourceKey,
-      },
-    ];
-  }
+  private mapTextResourceBindingsToExportFormat = (
+    textResourceBindings: ITextResourceBindings,
+  ): ExportTextResource[] => {
+    const result = [];
+    if (!textResourceBindings) return result;
+    Object.keys(textResourceBindings).map((textResourceBindingKey) => {
+      const textValues = this.getTextValues(
+        textResourceBindings[textResourceBindingKey],
+        (textResource) => textResource.id === textResourceBindings[textResourceBindingKey],
+      );
+      const textObject: ExportTextResource = {
+        id: textResourceBindings[textResourceBindingKey],
+        type: textResourceBindingKey,
+        text: textValues,
+      };
+      result.push(textObject);
+    });
+    return result;
+  };
 
-  return Object.keys(textResources).map((language) => ({
-    language: language,
-    value: textResources[language].find(findFunction)?.value ?? textResourceKey,
-  }));
-};
+  private getTextValues = (
+    textResourceKey: string,
+    findFunction?: (textResource: ITextResource) => boolean,
+  ): ExportTextResourceValue[] => {
+    if (this.defaultLanguage) {
+      return [
+        {
+          language: this.defaultLanguage,
+          value:
+            this.textResources[this.defaultLanguage].find(findFunction)?.value ?? textResourceKey,
+        },
+      ];
+    }
 
-const mapOptionsToExportFormat = (
-  options: IOption[],
-  textResources: ITextResources,
-  defaultLanguage: string,
-): ExportOption[] => {
-  if (!options) return undefined;
-  const result: ExportOption[] = options.map((option) => {
-    const textValues = getTextValues(
-      textResources,
-      option.label,
-      defaultLanguage,
-      (textResource) => textResource.id === option.label,
-    );
-    return {
-      value: option.value,
-      label: textValues,
-    };
-  });
-  return result;
-};
+    return Object.keys(this.textResources).map((language) => ({
+      language: language,
+      value: this.textResources[language].find(findFunction)?.value ?? textResourceKey,
+    }));
+  };
+
+  private mapOptionsToExportFormat = (options: IOption[]): ExportOption[] => {
+    if (!options) return undefined;
```

Comment:
Since TypeScript is indicating that options cannot be `null` or `undefined`, should we set an empty array as the default value? This way, the return value would be an empty array if no options are provided, aligning better with the expected return type. Otherwise we can change the  `options: IOption[] | undefined` and the return value to `ExportOption[] | undefined`. 

```suggestion
  private mapOptionsToExportFormat = (options: IOption[] = []): ExportOption[] => {
```

Reply:
Changed this, and the check for determining if a component has options (comment below), so that the undefined-check is done before this method is called 👍

Reply:
Nice!